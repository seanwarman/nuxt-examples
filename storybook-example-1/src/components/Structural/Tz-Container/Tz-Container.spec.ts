import { shallowMount, VueWrapper } from '@vue/test-utils';
import Container from './Tz-Container.vue';
import { describe, beforeEach, it, expect } from 'vitest';

const message = 'Container test for vitest update';
let wrapper: VueWrapper;

describe('Tz-Container.vue setup', () => {
  beforeEach(() => {
    wrapper = shallowMount(Container, {
      props: {
        styles: {
          width: 'medium'
        }
      },
      slots: {
        default: message
      }
    });
  });

  it('is a small container', () => {
    wrapper = shallowMount(Container, {
      props: {
        styles: {
          width: 'small'
        }
      }
    });

    expect(wrapper.classes()).toContain('max-w-small');
  });

  it('is a medium container', () => {
    wrapper = shallowMount(Container, {
      props: {
        styles: {
          width: 'medium'
        }
      }
    });

    expect(wrapper.classes()).toContain('max-w-medium');
  });

  it('is a large container', () => {
    wrapper = shallowMount(Container, {
      props: {
        styles: {
          width: 'large'
        }
      }
    });

    expect(wrapper.classes()).toContain('max-w-large');
  });

  it('is a full width container', () => {
    wrapper = shallowMount(Container, {
      props: {
        styles: {
          width: 'fluid'
        }
      }
    });

    expect(wrapper.classes()).toContain('max-w-full');
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  it('renders container content', () => {
    expect(wrapper.text()).toMatch(message);
  });
  it('sets a base class', () => {
    expect(wrapper.classes()).toContain('tz-container');
  });
});
