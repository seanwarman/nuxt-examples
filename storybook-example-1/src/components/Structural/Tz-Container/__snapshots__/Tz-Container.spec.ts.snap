// Vitest Snapshot v1

exports[`Tz-Container.vue setup > renders correctly 1`] = `
<div
  class="tz-container container mx-auto min-h-default max-w-medium"
>
  
  Container test for vitest update
  
</div>
`;
